### PR TITLE
Feat/alert/alertlists vm integration

### PR DESCRIPTION
**Pull request by agonzalez-r**
# Integrate Alert VM to AlertLists

## Description
This PR closes issue #17 and [#119](https://github.com/PeriodPals/periodpals/issues/119). 
It passes the `AlertViewModel` as parameter to `AlertListsScreen()` from `PeriodPalsApp()` in `MainActivity`.
It then integrates the Alert View Model to `AlertListsScreen()`, which now directly fetches the real data from repository for `myAlertList` and `palsAlertList`. UI was already amazingly done. 

**Note**: I had to add a private function `formatAlertTime()` to facilitate the parsing of the `alert.createdAt` value, because the previous expected format wasn't right.  

#### Added
None

#### Modified
- `app/src/main/java/com/android/periodpals/MainActivity.kt`
- `app/src/main/java/com/android/periodpals/ui/alert/AlertLists.kt`
- `app/src/main/java/com/android/periodpals/ui/alert/AlertLists.kt`

#### Removed
None

## Dependencies Added
None

## Testing
I adapted `AlertListsTests` by : 
- mocking AlertViewModel and AuthenticationViewModel
- refactoring the calls to `AlertListsScreen()`
- correcting the format of `location` and `createdAt` parameters of the mock Alerts
- mocking the response of the calls to `authenticationViewModel.authUserData`, `alertViewModel.alerts`, `alertViewModel.myAlerts` and `alertViewModel.palAlerts` depending on each tests

## Screenshots (if applicable)
Screenshot 2024-11-28 at 22 07 57
Screenshot 2024-11-28 at 22 08 05

### CODE REVIEW

**Review by coaguila — APPROVED** [review]
lgtm

File: app/src/main/java/com/android/periodpals/MainActivity.kt
Comment (francelu) [inline app/src/main/java/com/android/periodpals/MainActivity.kt:1]: LGTM. Changes are kept simple, consistent with existing code !

File: app/src/main/java/com/android/periodpals/ui/alert/AlertLists.kt [hunk 1]
```diff
@@ -187,12 +210,7 @@ fun AlertListsScreen(
  */
 @Composable
 private fun MyAlertItem(alert: Alert) {
-  // TODO: Change the logic about alert.id being null when implementing the AlertViewModel
-  if (alert.id == null) {
-    Log.d(LOG_TAG, "Alert id is null")
-    return
-  }
-  val idTestTag = alert.id
+  val idTestTag = alert.id!!
```
Comment (francelu) [inline app/src/main/java/com/android/periodpals/ui/alert/AlertLists.kt:213]: Why did you remove the null check here and not in pals alert item ?

File: app/src/main/java/com/android/periodpals/ui/alert/AlertLists.kt
Comment (francelu) [inline app/src/main/java/com/android/periodpals/ui/alert/AlertLists.kt:1]: Good, very good ! I like the `formatAlertTime()` function. It makes the code easier and lighter to read.

File: app/src/test/java/com/android/periodpals/ui/alert/AlertListsScreenTest.kt
Comment (francelu) [inline app/src/test/java/com/android/periodpals/ui/alert/AlertListsScreenTest.kt:1]: LGTM ! Changes are simple and consistent !

File: app/src/main/java/com/android/periodpals/ui/alert/AlertLists.kt [hunk 2]
```diff
@@ -86,18 +93,34 @@ private enum class AlertListsTab {
...
- *   for testing purposes. TODO: replace by the alertVM when implemented.
+ * @param alertViewModel The view model for managing alert data.
  */
 @Composable
 fun AlertListsScreen(
     navigationActions: NavigationActions,
-    myAlertsList: List<Alert> = emptyList(),
-    palsAlertsList: List<Alert> = emptyList(),
+    alertViewModel: AlertViewModel,
+    authenticationViewModel: AuthenticationViewModel
```
Comment (francelu) [inline app/src/main/java/com/android/periodpals/ui/alert/AlertLists.kt:102]: Missing authentication VM documentation.

**Review by francelu — APPROVED** [review]
## Summary
Changes are simple and consistent with existing code ! Good job ! Nothing to add.

## Important
#### Code Quality
- Code is readable and consistent with existing code. It is kept simple and easy to follow. Good job !

#### Functionality
- As we can see on the screenshots, code works as expected ! Great !

#### Performance
- Unchanged.

#### Security
- Unchanged.

## Testing
- Good adaptation of changes in existing tests !

## Documentation
- One missing, see comment.

## Suggestions
- See code comments.

## Steps before PR merged
Reply / resolve code comments :) Else all gooood ! Good job !! Have some well earned rest !!

File: app/src/main/java/com/android/periodpals/ui/alert/AlertLists.kt (hunk 1 above)
Comment (agonzalez-r) [reply app/src/main/java/com/android/periodpals/ui/alert/AlertLists.kt:213]: the `alert.id`  cannot be null (changes I did to Alert data class in #247). I'll make sure to remove the other check, I think I missed it
